src/events/processCollectedSignatures/index.js

```diff
@@ -52,20 +70,37 @@ function processCollectedSignaturesBuilder(config) {
 
           let gasEstimate
           try {
-            gasEstimate = await foreignBridge.methods
-              .executeSignatures(v, r, s, message)
-              .estimateGas()
+            gasEstimate = await estimateGas({
+              foreignBridge,
+              validatorContract,
+              v,
+              r,
+              s,
+              message,
+              numberOfCollectedSignatures: NumberOfCollectedSignatures,
+              address: VALIDATOR_ADDRESS
+            })
           } catch (e) {
-            if (e.message.includes('Invalid JSON RPC response')) {
+            if (e instanceof HttpListProviderError) {
               throw new Error(
-                `RPC Connection Error: executeSignatures Gas Estimate cannot be obtained.`
+                'RPC Connection Error: submitSignature Gas Estimate cannot be obtained.'
+              )
+            } else if (e instanceof AlreadyProcessedError) {
+              logger.info(
+                { eventTransactionHash: colSignature.transactionHash },
+                `Already processed CollectedSignatures ${colSignature.transactionHash}`
               )
+              return
+            } else if (e instanceof IncompatibleContractError) {
+              logger.fatal(`The contract is not compatible: ${e.message}`)
+              process.exit(10)
```

Comment:
So far I see only one reason when `IncompatibleContractError` appears - when number of signatures collected on `Home` side less than number of signatures expected by the contract on `Foreign` side. It could be valid scenario when number of required signatures was changed and the bridge instance is trying to re-handle some messages (e.g.  database of handled blocks was rolled back). The bridge must not crash here, it is incorrect behavior.

Reply:
What would be the correct behavior here? If we ignore the message, it will never be processed. If we just throw, it will keep retrying forever.

Reply:
For scenario when number of signatures collected on `Home` side less than number of signatures expected by the contract on `Foreign` side we need report about this in the log and handle it by the same as we handle e.g. `AlreadySigned`.

Reply:
> So, in this case, we log the occurrence and don't process the message.

Correct.

>  I guess it's also something that could be monitored.

Yes, we should see the corresponding record at least in the logs.

Reply:
Done.